pkg/controller/dataimportcron-controller_test.go

```diff
@@ -273,6 +273,42 @@ var _ = Describe("All DataImportCron Tests", func() {
 			Expect(len(dvList.Items)).To(Equal(0))
 		})
 
+		It("Should reconcile only if DataSource is not labeled by another existing DIC", func() {
+			cron = newDataImportCron(cronName)
+			reconciler = createDataImportCronReconciler(cron)
+
+			shouldReconcile, err := reconciler.shouldReconcileCron(context.TODO(), cron)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(shouldReconcile).To(Equal(true))
+
+			_, err = reconciler.Reconcile(context.TODO(), cronReq)
+			Expect(err).ToNot(HaveOccurred())
+
+			dataSource = &cdiv1.DataSource{}
+			err = reconciler.client.Get(context.TODO(), dataSourceKey(cron), dataSource)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(dataSource.Labels[common.DataImportCronLabel]).To(Equal(cron.Name))
+
+			cron1 := newDataImportCron(cronName + "1")
+			shouldReconcile, err = reconciler.shouldReconcileCron(context.TODO(), cron1)
```

Comment:
Can you check that the event is fired here? The reconciler should have an event slice you can check.

Reply:
added